logstash-core/lib/logstash/runner.rb

```diff
@@ -440,6 +442,21 @@ def execute
     @log_fd.close if @log_fd
   end # def self.main
 
+  def running_as_root
+    running_as_root = setting("on_superuser")
+    running_as_root_allowed_options = { "allow" => "ALLOW", "warn" => "WARN", "block" => "BLOCK" }
+    if running_as_root_allowed_options.has_value?(running_as_root) == false
+      logger.warn("Allowed options on on_superuser are WARN, BLOCK and ALLOW. Falling back to default WARN.")
+      running_as_root = running_as_root_allowed_options["warn"]
+    end
+    if Process.euid() == 0 && running_as_root.eql?(running_as_root_allowed_options["allow"]) == false
+      deprecation_message_on_root = "NOTICE: Running logstash as root is not recommended because of security reasons. This will be deprecated in the future."
+      error_message_on_root = "Logstash cannot be run as root."
+      logger.warn(deprecation_message_on_root) if running_as_root.eql?(running_as_root_allowed_options["warn"])
+      raise(RuntimeError, error_message_on_root) if running_as_root.eql?(running_as_root_allowed_options["block"])
```

Comment:
We can log a deprecation log using the deprecation logger instead `logger.warn`. This way the logging entry ends up the right log file (i.e. `logs/logstash-deprecation.log` vs `logs/logstash-plain.log`).
Also we should add a call to action in the deprecation notice to guide to user towards what they need to do:

```suggestion
      if running_as_root == "WARN"
        deprecation_logger.deprecated("NOTICE: Running Logstash as root is not recommended and won't be allowed in the future. Set 'on_superuser' to 'BLOCK' to avoid startup errors in future releases.".)
      else # BLOCK
        raise(RuntimeError, "Logstash cannot be run as root.") 
      end
```

Reply:
I have limited visibility to `deprecation logs` and one question though.
I don't see any types of logs are appeared on console when I use `deprecation_logger` and message will be logged into `logstash-deprecation.log` file. I wonder how customers are able to easily see this log to take an action?